toolchain/check/handle_class.cpp

```diff
@@ -181,10 +181,13 @@ static auto BuildClassDecl(Context& context, Parse::AnyClassDeclId node_id,
   auto introducer =
       context.decl_introducer_state_stack().Pop<Lex::TokenKind::Class>();
   CheckAccessModifiersOnDecl(context, introducer, parent_scope_inst);
-  LimitModifiersOnDecl(context, introducer,
-                       KeywordModifierSet::Access | KeywordModifierSet::Extern |
-                           (is_definition ? KeywordModifierSet::Class
-                                          : KeywordModifierSet::None));
+  auto accepted_modifiers =
+      KeywordModifierSet::Access | KeywordModifierSet::Extern;
+  auto invalid_unless_definition_modifiers = KeywordModifierSet::None;
+  (is_definition ? accepted_modifiers : invalid_unless_definition_modifiers) |=
+      KeywordModifierSet::Class;
+  LimitModifiersOnDecl(context, introducer, accepted_modifiers,
+                       invalid_unless_definition_modifiers);
```

Comment:
I find this a bit hard to read. I wonder if it would be simpler to introduce another function for this case? Maybe something like:

```suggestion
  LimitModifiersOnDecl(context, introducer,
                       KeywordModifierSet::Access | KeywordModifierSet::Extern |
                       KeywordModifierSet::Class);
  if (!is_definition) {
    LimitModifiersNotDefinition(context, introducer,
                         KeywordModifierSet::Access | KeywordModifierSet::Extern);
  }
```

The only difference between the two would be the diagnostic issued, so they could mostly share implementation.

Reply:
Thanks for the suggestion, yeah - that does simplify things greatly! (done)